napari/_qt/_tests/test_qt_viewer.py

```diff
@@ -783,22 +798,28 @@ def test_axes_labels(make_napari_viewer):
 def qt_viewer(qtbot):
     qt_viewer = QtViewer(ViewerModel())
     qt_viewer.show()
-    qt_viewer.resize(400, 400)
+    qt_viewer.resize(460, 460)
     yield qt_viewer
     qt_viewer.close()
+    qt_viewer._instances.clear()
     del qt_viewer
-    qtbot.wait(50)
-    gc.collect()
 
 
+# @pytest.mark.xfail(reason="Fails on CI, but not locally")
 @skip_local_popups
 @pytest.mark.parametrize('direct', [True, False], ids=["direct", "auto"])
 def test_thumbnail_labels(qtbot, direct, qt_viewer: QtViewer):
     # Add labels to empty viewer
     layer = qt_viewer.viewer.add_labels(np.array([[0, 1], [2, 3]]), opacity=1)
     if direct:
         layer.color = {0: 'red', 1: 'green', 2: 'blue', 3: 'yellow'}
-    qtbot.wait(100)
+    else:
+        layer.num_colors = 49
+    qt_viewer.viewer.reset_view()
+    # this test is fragile as reset view is not always working
+    # I do not know how to enforce it.
```

Comment:
Is it possible that calling `QApplication.processEvents()` here would help? Or is it not feasible when using qtbot?

Reply:
I have added `QApplication.processEvents`. Now we could observe if problem do not happens

Reply:
There's an (I think) unrelated failure in the latest run:

```
_________________________________ test_polygon _________________________________

    def test_polygon():
        """Test creating Shape with a random polygon."""
        # Test a single six vertex polygon
        data = np.array(
            [
                [10.97627008, 14.30378733],
                [12.05526752, 10.89766366],
                [8.47309599, 12.91788226],
                [8.75174423, 17.83546002],
                [19.27325521, 7.66883038],
                [15.83450076, 10.5778984],
            ]
        )
        shape = Polygon(data)
        np.testing.assert_array_equal(shape.data, data)
        assert shape.data_displayed.shape == (6, 2)
        assert shape.slice_key.shape == (2, 0)
        # should get few triangles
        assert shape._edge_vertices.shape == (16, 2)
>       assert shape._face_vertices.shape == (8, 2)
E       assert (6, 2) == (8, 2)
E         At index 0 diff: 6 != 8
E         Full diff:
E         - (8, 2)
E         ?  ^
E         + (6, 2)
E         ?  ^
```

I have absolutely no understanding what could cause this test to fail randomly...

Reply:
Oh, I just saw that that is part of these commits... Crazy! Has there been a change in main or to constraints? 🤔

Reply:
What you see is my attempt to work around this problem. 
I just ran test locally and copied the points coordinates. 

I need to investigate why this is passing on tox and not on pip install.

Reply:
It is because I add `optional` to `all` extras and it starts using the triangles library instead of the vispy fallback.